### PR TITLE
pd-worker: fix get_members_opt blocking

### PR DESCRIPTION
**Pull request by NingLin-P**
Signed-off-by: linning <linningde25@gmail.com>

### What problem does this PR solve?

Issue Number: #7124

Problem Summary:

When pd-server got killed, pd-worker will try to reconnect pd and issue `get_members_opt` request but this may never return even timeout reached , and pd worker can not handle any new task. 

What's Changed:

Use `get_members_async_opt` and add external timeout mechanism.

### Related changes

No
### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by BusyJay** [comment]
So the second time it reconnects succeeds?

**Comment by siddontang** [comment]
> request but this may never return even timeout reached

why never return?

**Comment by NingLin-P** [comment]
> > request but this may never return even timeout reached
> 
> why never return?

This pr is a workaround for now, the root cause still need more investigating.

**Comment by NingLin-P** [comment]
> So the second time it reconnects succeeds?

When the pd restart and the underlying grpc connect established, it will reconnect succeed, this pr is aims to prevent pd worker from blocking and can retry later when new task arrived or  reconnect interval elapsed.

File: components/pd_client/src/util.rs [hunk 1]
```diff
@@ -435,9 +435,20 @@ fn connect(
...
     let option = CallOption::default().timeout(Duration::from_secs(REQUEST_TIMEOUT));
-    match client.get_members_opt(&GetMembersRequest::default(), option) {
-        Ok(resp) => Ok((client, resp)),
-        Err(e) => Err(Error::Grpc(e)),
+    let f = client
+        .get_members_async_opt(&GetMembersRequest::default(), option)
+        .map_err(Error::Grpc)?;
+    let delay = GLOBAL_TIMER_HANDLE
+        .delay(Instant::now() + Duration::from_secs(REQUEST_TIMEOUT))
+        .then(|_| {
```
Comment (5kbpers) [inline components/pd_client/src/util.rs]: Could we log the error here?

**Comment by NingLin-P** [comment]
Close this pr since #7253 fixed issue #7124 .

**Pull request closed without merging**